tests/test_equilibrium.py

```diff
@@ -464,3 +466,156 @@ def test_backward_compatible_load_and_resolve():
+        modes_Z=np.array([[0, 0], [-1, 0]]),
+        sym=True,
+        NFP=1.0,
+    )
+
+    pressure = PowerSeriesProfile([1.5e5, 0, -1.5e5])
+    iota = PowerSeriesProfile([0.5, 0, -0.2])
+
+    eq = Equilibrium(
+        L=10, M=10, N=0, surface=surf, Psi=0.1, pressure=pressure, iota=iota
+    )
+
+    eq, _ = eq.solve(objective="force", ftol=1e-3, gtol=1e-6, maxiter=30, verbose=3)
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    psi = 0.25  # rho^2 (or normalized psi)
+    alpha = 0
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    data_eq = eq.compute(eq_keys)
+
+    iotas = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota"])
+    shears = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota_r"])
+
+    N = int((2 * eq.M_grid) * 4 + 1)
+
+    zeta = np.linspace(-1.0 * np.pi / iotas, 1.0 * np.pi / iotas, N)
+    theta_PEST = alpha * np.ones(N, dtype=int) + iotas * zeta
+
+    coords1 = np.zeros((N, 3))
+    coords1[:, 0] = np.sqrt(psi) * np.ones(N, dtype=int)
+    coords1[:, 1] = theta_PEST
+    coords1[:, 2] = zeta
+
+    # Creating a grid along a field line
+    c1 = eq.compute_theta_coords(coords1)
+    grid = Grid(c1, sort=False)
+
+    data_keys = [
+        "kappa",
+        "|grad(psi)|^2",
+        "grad(|B|)",
+        "grad(alpha)",
+        "grad(psi)",
+        "B",
+        "grad(|B|)",
+        "iota",
+        "|B|",
+        "B^zeta",
+        "cvdrift0",
+        "cvdrift",
+        "gbdrift",
+    ]
+
+    data = eq.compute(data_keys, grid=grid, override_grid=False)
+
+    psib = data_eq["psi"][-1]
+
+    # signs
+    sign_psi = psib / np.abs(psib)
+    sign_iota = iotas / np.abs(iotas)
+
+    # normalizations
+    Lref = data_eq["a"]
+    Bref = 2 * np.abs(psib) / Lref**2
+
+    modB = data["|B|"]
+    bmag = modB / Bref
+
+    x = Lref * np.sqrt(psi)
+    shat = -x / iotas * shears / Lref
+
+    grad_psi = data["grad(psi)"]
+    grad_alpha = data["grad(alpha)"]
+
+    iota = data["iota"]
+
+    gradpar = Lref * data["B^zeta"] / modB
+
+    gds21 = -sign_iota * np.array(dot(grad_psi, grad_alpha)) * shat / Bref
+
+    gbdrift = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_alpha))
+    gbdrift *= -sign_psi * 2 * Bref * Lref**2 / modB**3 * np.sqrt(psi)
+
+    cvdrift = (
+        -sign_psi
+        * 2
+        * Bref
+        * Lref**2
+        * np.sqrt(psi)
+        * dot(cross(data["B"], data["kappa"]), grad_alpha)
+        / modB**2
+    )
+
+    cvdrift0 = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_psi))
+    cvdrift0 *= sign_iota * sign_psi * shat * 2 / modB**3 / np.sqrt(psi)
+
+    cvdrift_2 = data["cvdrift"]
+    gbdrift_2 = data["gbdrift"]
+
+    ## Comparing coefficient calculation here with coefficients from compute
+    cvdrift_2 = -2 * sign_psi * Bref * Lref**2 * np.sqrt(psi) * data["cvdrift"]
+    gbdrift_2 = -2 * sign_psi * Bref * Lref**2 * np.sqrt(psi) * data["gbdrift"]
+
+    np.testing.assert_allclose(gbdrift, gbdrift_2, atol=1e-4)
```

Comment:
in most other tests we've been trying to use `rtol` instead of `atol` where possible, mainly so when reading the tests we can get a sense of the expected % error without having to know the scale of the data

Reply:
I use atol because these quantities cross the y = 0 line multiple time and rtol breaks at points where we have 0/0 or 1/0 type situation.

Reply:
Using both atol and rtol can alleviate this, so that the atol can be made like 1e-8 or something very small, and that only activates when the data is crossing zero